### PR TITLE
Hyperlink social media

### LINKED ISSUE #79: Hyperlink social media
As a user, I want to be able to click on the social media link and open it on the browser, so that I would not have to manually type out or copy-paste the links.

### PR DESCRIPTION
**Pull request by neyapraveen**
Create a hyperlink for social media added. 
Fixes #79

### CODE REVIEW

**Comment by neyapraveen** [comment]
Hyperlink not compatible with multiple social media links yet.

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 1]
```diff
@@ -52,16 +53,28 @@ public PersonCard(Person person, int displayedIndex) {
...
+                .sorted()
+                .collect(Collectors.joining(", "));
         tutorials.setText(tutorialsText);
         person.getSocialMediaLinks().stream()
                 .sorted(Comparator.comparing(sm -> sm.socialMediaLink))
                 .forEach(sm -> {
-                    Label label = new Label(sm.socialMediaLink);
-                    label.setStyle("-fx-font-size: 13px; -fx-text-fill: white; -fx-font-family: 'Segoe UI Semibold';");
-                    socialMediaLinks.getChildren().add(label);
+                    Hyperlink hyperlink = new Hyperlink(sm.socialMediaLink);
```
Comment (neyapraveen) [inline src/main/java/seedu/address/ui/PersonCard.java:63]: Needs to work with multiple links which appear on the same line

Comment (neyapraveen) [reply src/main/java/seedu/address/ui/PersonCard.java:65]: Will implement that in another PR as it does not affect functionality compared to our old code. Does not throw any errors either.

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 2]
```diff
@@ -52,16 +53,37 @@ public PersonCard(Person person, int displayedIndex) {
...
         description.setText(person.getDescription().value);
         String tutorialsText = person.getTutorials().stream()
-            .map(t -> "T" + t.getValue())
-            .sorted()
-            .collect(Collectors.joining(", "));
+                .map(t -> "T" + t.getValue())
+                .sorted()
+                .collect(Collectors.joining(", "));
         tutorials.setText(tutorialsText);
+        int initialHyperlinksCount = socialMediaLinks.getChildren().size(); // Get the initial count
```
Comment (wnchan) [inline src/main/java/seedu/address/ui/PersonCard.java:60]: I think need to add test case for the new changes.

**Review by wnchan — COMMENTED** [review]
LGTM! just need to add test cases for the hyperlinks.

File: src/main/java/seedu/address/ui/PersonCard.java (hunk 2 above)
Comment (neyapraveen) [reply src/main/java/seedu/address/ui/PersonCard.java:66]: Yep done!

**Review by Bearypop — COMMENTED** [review]
Nice! LGTM

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -298,8 +298,8 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 **MSS**
 
-1.  Student requests to list students
-2.  StudentConnect shows a list of students
+1.  Student requests to list students.
```
Comment (PearlynnT) [inline docs/DeveloperGuide.md:301]: Thanks for fixing the punctuation in the DG

File: src/test/java/seedu/address/ui/JavaFxInitializer.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        /**
+         * Perform cleanup logic if needed.
+         * This method can be overridden in subclasses to add custom cleanup logic.
+         */
+        public void cleanup() {
+            // Call any cleanup logic here if needed
+        }
+
+        /**
+         * Get the instance of the JavaFX initialization class.
```
Comment (PearlynnT) [inline src/test/java/seedu/address/ui/JavaFxInitializer.java:56]:
```suggestion
         * Gets the instance of the JavaFX initialization class.
```

File: src/test/java/seedu/address/ui/JavaFxInitializer.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+        @Override
+        public void start(Stage primaryStage) {
+            // Save the instance for later cleanup
+            instance = this;
+            primaryStage.hide();
+        }
+
+        /**
+         * Perform cleanup logic if needed.
```
Comment (PearlynnT) [inline src/test/java/seedu/address/ui/JavaFxInitializer.java:48]:
```suggestion
         * Performs cleanup logic if needed.
```

**Review by PearlynnT — COMMENTED** [review]
Thanks for adding the hyperlink, which is an impt feature

File: src/test/java/seedu/address/ui/JavaFxInitializer.java (hunk 5 above)
Comment (neyapraveen) [reply src/test/java/seedu/address/ui/JavaFxInitializer.java]: Fixed both. Thanks!

**Pull request merged**